### PR TITLE
Update `i18n.formatCurrency` to use `form: 'auto'` by default

### PR DESCRIPTION
**Pull request by trishrempel**
## Description

Related to https://github.com/Shopify/quilt/issues/2304

Currently, if you do not provide the `form` option to `i18n.formatCurrency`, then its behaviour reverts to the legacy (deprecated) `formatCurrency()`, which is a convenience function that simply _auto-assigns_ the `as` option to `currency` and calls `formatNumber()`. 

Note that this will resemble `form: 'short'`, but will sometimes extend the symbol with extra information depending on the browser's implementation of `Intl.NumberFormat` and the locale in use. For example, `formatCurrency(1.25, {currency: 'CAD'})` may return `$ 1.25`, or it might return `CA$ 1.25`.

There are many instances in Web that call `i18n.formatCurrency(amount)` without the `form` option, and so are not displaying currency formats according to [Polaris localized currency formatting guidelines](https://polaris.shopify.com/foundations/formatting-localized-currency).

This pull request changes the default behaviour to use `form: 'auto'` when the `form` option is not provided.

### CODE REVIEW

**Review by BPScott — APPROVED** [review]
Love this!

Big fan of making sure we have sensible defaults!

When you're happy go forth and squash and merge.

**Comment by sunical** [comment]
@trishrempel, I didn't find the test coverage for this scenario: when `form` is omitted, default to `form: 'auto'`

File: packages/react-i18n/src/tests/i18n.test.ts [hunk 1]
```diff
@@ -784,16 +784,48 @@ describe('I18n', () => {
   });
 
   describe('#formatCurrency()', () => {
+    const amount = 1234;
     const currency = 'USD';
 
-    it('formats the number as a currency', () => {
+    it('formats short when there is no currency in the options', () => {
```
Comment (trishrempel) [inline packages/react-i18n/src/tests/i18n.test.ts:790]: @sunical These tests are copied from the `form:auto` tests just below this - they confirm that not passing the `form` option produces the same behaviour as `form:auto`.

**Pull request merged**